.github/workflows/sizeComparison.yml

```diff
@@ -0,0 +1,66 @@
+name: sizeComparison
+
+on:
+  push:
+    branches:
+      - main*
+      - fluent2*
+  pull_request:
+    branches:
+      - main*
+      - fluent2*
+
+jobs:
+  archiveHeadBuild:
+    runs-on: macos-12
+    strategy:
+      fail-fast: false
+    steps: 
+    - uses: actions/checkout@v3 # Checkout repo at head sha
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+    - name: Switch to current version of Xcode
+      run: scripts/xcode_select_current_version.sh
+    - name: scripts/xcodebuild_wrapper.sh 'ios_device_build Demo.Development Release build -archivePath $GITHUB_WORKSPACE/headBuild archive' # Build and archive release device build
+      run: scripts/xcodebuild_wrapper.sh ios_device_build Demo.Development Release build -archivePath $GITHUB_WORKSPACE/${{ github.event.pull_request.head.sha }} archive
+    - name: Create size file # Get size of demo app in "kilobytes" (2*512B blocks) and store in a file with the hash as the name
+      run: du -sk $GITHUB_WORKSPACE/${{ github.event.pull_request.head.sha }}.xcarchive/Products/Applications/FluentUI.Demo.app | cut -f1 >> $GITHUB_WORKSPACE/${{ github.event.pull_request.head.sha }}
+    - name: Upload head size artifact # Upload the file with just the size, ie, 36692
+      uses: actions/upload-artifact@v3
+      with:
+        name: ${{ github.event.pull_request.head.sha }}
+        path: ${{ github.event.pull_request.head.sha }}
+  archiveBaseBuild:
+    runs-on: macos-12
+    strategy:
+      fail-fast: false
+    steps: 
+    - uses: actions/checkout@v3
```

Comment:
If we can't create a reusable job or composite action, I'd love to see more symmetry and use `with: ref: ${{ github.event.pull_request.base.sha }}` So that I don't have to think that BASE contains the 'merged commit' and HEAD is what the PR is based on... Where locally HEAD before submitting it contains your changes... 

Maybe even name then different than base and head... i.e. `withchanges` and `withoutchanges` or `before` and `after`.